easy_profiler_core/event_trace_win.cpp

```diff
@@ -244,7 +244,12 @@ namespace profiler {
                         {
                             pinfo->name.reserve(pinfo->name.size() + 2 + len);
                             pinfo->name.append(" ", 1);
+#if UNICODE
+                            std::wstring wbuf(buf);
```

Comment:
Maybe it would be better to modify `struct ProcessInfo` instead of function code?
```cpp
struct ProcessInfo {
#if UNICODE
        std::wstring     name;
#else
        std::string      name;
#endif
        processid_t    id = 0;
        int8_t      valid = 0;
    };
```

Reply:
Or
```cpp
#if UNICODE
        using ProcessName = std::wstring;
#else
        using ProcessName = std::string;
#endif

struct ProcessInfo {
        ProcessName name;
        processid_t    id = 0;
        int8_t      valid = 0;
    };
```

Reply:
I refactored ProcessInfo. There are multiple ways how to approach handling of wstring, so I'm not sure if my approach is the best one.  Please have a look. 
Your proposal would be a way, but in the end we want to convert it to a const char* to pass it on to the manager, so I thought we should just store it as std::string already and handle the unicode stuff during construction of the info string.
I also removed the property "id", because it's not used and simplified the flag "valid", because nowhere it's checked whether the initialization succeeded or not (i.e. setting valid to 1 or -1).